inference-engine/tests/functional/shared_test_classes/src/single_layer/matrix_nms.cpp

```diff
@@ -206,45 +255,37 @@ void MatrixNmsLayerTest::Compare(const std::vector<std::pair<ngraph::element::Ty
-        attrs.background_class, attrs.normalized, attrs.decay_function, targetDevice) = this->GetParam();
+    std::tie(shapes, inPrecisions, m_attrs.sort_result_type, m_attrs.output_type, topKParams, thresholdParams,
+        m_attrs.background_class, m_attrs.normalized, m_attrs.decay_function, targetDevice) = this->GetParam();
 
-    std::tie(attrs.nms_top_k, attrs.keep_top_k) = topKParams;
-    std::tie(attrs.score_threshold, attrs.gaussian_sigma, attrs.post_threshold) = thresholdParams;
-    std::tie(numBatches, numBoxes, numClasses) = inShapeParams;
-    auto realClasses = numClasses;
-    if (attrs.background_class >=0 && attrs.background_class <= numClasses) {
-        realClasses = realClasses - 1;
-    }
+    std::tie(m_attrs.nms_top_k, m_attrs.keep_top_k) = topKParams;
+    std::tie(m_attrs.score_threshold, m_attrs.gaussian_sigma, m_attrs.post_threshold) = thresholdParams;
 
-    maxOutputBoxesPerClass = 0;
-    if (attrs.nms_top_k >= 0)
-        maxOutputBoxesPerClass = std::min(numBoxes, static_cast<size_t>(attrs.nms_top_k));
-    else
-        maxOutputBoxesPerClass = numBoxes;
+    init_input_shapes(shapes);
 
-    maxOutputBoxesPerBatch  = maxOutputBoxesPerClass * realClasses;
-    if (attrs.keep_top_k >= 0)
-        maxOutputBoxesPerBatch =
-                std::min(maxOutputBoxesPerBatch, static_cast<size_t>(attrs.keep_top_k));
-    Precision paramsPrec, maxBoxPrec, thrPrec;
+    ElementType paramsPrec, maxBoxPrec, thrPrec;
     std::tie(paramsPrec, maxBoxPrec, thrPrec) = inPrecisions;
-
-    const std::vector<size_t> boxesShape{numBatches, numBoxes, 4}, scoresShape{numBatches, numClasses, numBoxes};
-    auto ngPrc = convertIE2nGraphPrc(paramsPrec);
-    auto params = builder::makeParams(ngPrc, {boxesShape, scoresShape});
-    auto paramOuts = helpers::convert2OutputVector(helpers::castOps2Nodes<op::Parameter>(params));
-    auto nms = std::make_shared<opset8::MatrixNms>(paramOuts[0], paramOuts[1], attrs);
-    auto nms_0_identity = std::make_shared<opset5::Multiply>(nms->output(0), opset5::Constant::create(element::f32, Shape{1}, {1}));
-    auto nms_1_identity = std::make_shared<opset5::Multiply>(nms->output(1), opset5::Constant::create(attrs.output_type, Shape{1}, {1}));
-    auto nms_2_identity = std::make_shared<opset5::Multiply>(nms->output(2), opset5::Constant::create(attrs.output_type, Shape{1}, {1}));
-    function = std::make_shared<Function>(OutputVector{nms_0_identity, nms_1_identity, nms_2_identity}, params, "NMS");
+    const auto params = ngraph::builder::makeDynamicParams(paramsPrec, inputDynamicShapes);
+    const auto paramOuts =
+            ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+    auto nms = std::make_shared<opset8::MatrixNms>(paramOuts[0], paramOuts[1], m_attrs);
+    if (targetDevice == CommonTestUtils::DEVICE_CPU) {
+        m_outStaticShape = false;
```

Comment:
since you add `m_outStaticShape` in tests maybe better create graph `NMS -> Result` if `m_outStaticShape == false`
and `NMS -> Multiply -> Result` if `m_outStaticShape == true`

same for `MultiClassNmsTests`

Reply:
Done.